### ISSUE TITLE
Disallow multiple Dimension per Fields 

### ISSUE THREAD

**Issue by camsaul** [metabase/metabase#27054]
Dimensions are one of the many ways we control Field remapping. See the explanations [here](https://github.com/metabase/metabase/blob/07c4d59174ddf254877dba40d5c6b3ec75cc966b/src/metabase/models/params/chain_filter.clj#L1) and [here](https://github.com/metabase/metabase/blob/07c4d59174ddf254877dba40d5c6b3ec75cc966b/src/metabase/query_processor/middleware/add_dimension_projections.clj) for more context.

The Dimension table, added in https://github.com/metabase/metabase/pull/5130, has a unique constraint on `(id, field_name)`, which implies the original intention was to allow multiple Dimensions per Field, as long as they had different names. The fact that we hydrate Field with `:dimensions` rather than `:dimension` also supports this theory, altho there is a comment saying this code is broken and only returns one dimension:

https://github.com/metabase/metabase/blob/07c4d59174ddf254877dba40d5c6b3ec75cc966b/src/metabase/models/field.clj#L273-L284

However, it looks like lots of our code is clearly written under the assumption that a Field would have either 0 or 1 dimensions, not 0 **or more**:

https://github.com/metabase/metabase/blob/07c4d59174ddf254877dba40d5c6b3ec75cc966b/src/metabase/query_processor/middleware/add_dimension_projections.clj#L63-L65

https://github.com/metabase/metabase/blob/07c4d59174ddf254877dba40d5c6b3ec75cc966b/src/metabase/models/params/chain_filter.clj#L520-L525

To summarize, in practice we do not actually support multiple Dimensions per Field, altho it looks like we originally intended to. Is this something we should fix, or can we make Dimensions be unique per Field? Once we decide on a course of action we should fix things as it seems this is probably a cause of flaky tests in CI if not of more serious bugs.

Related: https://github.com/metabase/metabase/issues/668

**Comment by camsaul** [comment]
[Discussed this with Sameer](https://metaboat.slack.com/archives/C01LQQ2UW03/p1670444787015729) and it sounds like the consensus is that we want to phase remappings entirely in the future in favor of models. So I think we can go ahead and make the mapping 1t1 here and fix tests that way. The approach I suggested in #668 should work here too

**Comment by camsaul** [comment]
Fixed by #27062
